final_prep/mod_zero_hero.rb

```diff
@@ -2,52 +2,121 @@
 
 # Declare two variables - hero_name AND special_ability - set to strings
 
+hero_name = "Catman"
+special_ability = "napping"
+
 # Declare two variables - greeting AND catchphrase
 #   greeting should be assigned to a string that uses interpolation to include the hero_name
 #   catchphrase should be assigned to a string that uses interpolation to include the special_ability
 
+greeting = "Hello, my name is #{hero_name}!"
+catchphrase = "No one is better at #{special_ability} than I am!"
+
 # Declare two variables - power AND energy - set to integers
 
+power = 8
+energy = 4
+
 # Declare two variables - full_power AND full_energy
 #   full_power should multiply your current power by 500
 #   full_energy should add 150 to your current energy
 
+full_power = power * 500
+full_energy = energy * 150
+
 # Declare two variables - is_human and identity_concealed - assigned to booleans
 
+is_human = true
+identity_concealed = false
 
 # Declare two variables - arch_enemies AND sidekicks
 #   arch_enemies should be an array of at least 3 different enemy strings
 #   sidekicks should be an array of at least 3 different sidekick strings
 
+arch_enemies = ["Dog Boy", "Super Fish", "Alarm Clocks"]
+sidekicks = ["Mouse Toy", "Soft Blanket", "Catnip"]
+
 # Print the first sidekick to your terminal
 
+print sidekicks[0]
+
 # Print the last arch_enemy to the terminal
 
+print arch_enemies[2]
+
 # Write some code to add a new arch_enemy to the arch_enemies array
 
+arch_enemies.push("Vacuum Cleaner")
+
 # Print the arch_enemies array to terminal to ensure you added a new arch_enemey
 
+print arch_enemies
+
 # Remove the first sidekick from the sidekicks array
 
+sidekicks.shift
+
 # Print the sidekicks array to terminal to ensure you added a new sidekick
 
+print sidekicks
+
 # Create a function called assess_situation that takes three arguments - danger_level, save_the_day, bad_excuse
 #   - danger_level should be an integer
-#   - save_the_day should be a string a hero would say once they save the day 
+#   - save_the_day should be a string a hero would say once they save the day
 #   - bad_excuse should be a string a hero would say if they are too afraid of the danger_level
 
+def assess_situation(danger_level, save_the_day, bad_excuse)
+  @danger_level = danger_level
+  @save_the_day = save_the_day
+  @bad_excuse = bad_excuse
+end
+
+def save_the_day
```

Comment:
You don't need to define `save_the_day`, `bad_excuse` and `danger_level` as functions - they are variables passed in as arguments to the `assess_situation` function.